rsconnect/bundle.py

```diff
@@ -1241,6 +1305,59 @@ def write_notebook_manifest_json(
     return exists(join(directory, environment.filename))
 
 
+def write_voila_manifest_json(
+    path: str,
+    entrypoint: str,
+    environment: Environment,
+    app_mode: AppMode = AppModes.JUPYTER_VOILA,
+    extra_files: typing.List[str] = None,
+    excludes: typing.List[str] = None,
+    force_generate: bool = True,
+    image: str = None,
+) -> bool:
+    """
+    Creates and writes a manifest.json file for the given path.
+
+    :param path: the file, or the directory containing the files to deploy.
+    :param entry_point: the main entry point for the API.
+    :param environment: the Python environment to start with.  This should be what's
+    returned by the inspect_environment() function.
+    :param app_mode: the application mode to assume.  If this is None, the extension
+    portion of the entry point file name will be used to derive one. Previous default = None.
+    :param extra_files: any extra files that should be included in the manifest. Previous default = None.
+    :param excludes: a sequence of glob patterns that will exclude matched files.
+    :param force_generate: bool indicating whether to force generate manifest and related environment files.
+    :param image: the optional docker image to be specified for off-host execution. Default = None.
+    :return: whether the manifest was written.
+    """
+    mimetypes.add_type("text/ipynb", ".ipynb")
```

Comment:
This mimetype should be a constant - it appears several times in the code as a literal.